packages/headless/src/controllers/facet-manager/headless-facet-manager.ts

```diff
@@ -1,31 +1,63 @@
 import {Engine} from '../../app/headless-engine';
 import {SearchSection} from '../../state/state-sections';
 import {sortFacets} from '../../utils/facet-utils';
-import {buildController} from '../controller/headless-controller';
+import {buildController, Controller} from '../controller/headless-controller';
 
-export type FacetManagerPayload<T> = {
+/**
+ * A facet payload object to be sorted by the manager.
+ */
+export interface FacetManagerPayload<T> {
+  /**
+   * A unique string identifying a facet.
+   */
   facetId: string;
+
+  /**
+   * The payload to associate with the facetId. This can be anything e.g., a DOM element, JSX, a string.
+   */
   payload: T;
-};
+}
+
+/**
+ * The `FacetManager` controller helps reorder facets to match the most recent search response.
+ */
+export interface FacetManager extends Controller {
+  /**
+   * Sorts the facets to match the order in the most recent search response.
+   *
+   * @param facets - An array of facet payloads to sort.
+   * @returns A sorted array.
+   */
+  sort<T>(facets: FacetManagerPayload<T>[]): FacetManagerPayload<T>[];
 
-export type FacetManager = ReturnType<typeof buildFacetManager>;
-export type FacetManagerState = FacetManager['state'];
+  /**
+   * The state of the `FacetManager` controller.
+   * */
+  state: FacetManagerState;
+}
+
+export interface FacetManagerState {
+  /**
+   * The facet ids sorted in the same order as the latest response.
+   */
+  facetIds: string[];
+}
 
-export function buildFacetManager(engine: Engine<SearchSection>) {
+/**
+ * Creates a `FacetManger` instance.
+ *
+ * @param engine - The headless engine.
+ */
+export function buildFacetManager(engine: Engine<SearchSection>): FacetManager {
   const controller = buildController(engine);
 
   return {
     ...controller,
 
-    /** Sorts the facets to match the order in the most recent search response.
-     * @param FacetManagerPayload[] An array of facet payloads to sort.
-     * @returns FacetManagerPayload[].
-     */
     sort<T>(facets: FacetManagerPayload<T>[]) {
```

Comment:
Optionally, the parameter type here could be removed.

Reply:
Hmm, if I remove the generic `T` param, I have a ts error with the `T` passed to the `FacetManagerPayload` interface. Is it not the case?